### PR TITLE
 #159965275 Implementing sharing of articles

### PR DESCRIPTION
**Pull request by mikenthiwa**
#### What does this PR do?
 * Describes implementation of sharing articles via twitter, facebook and email.
#### Description of Task to be completed?
 * Users are able to share articles through facebook, twitter and email. 
#### How should this be manually tested?
 $ Clone this repo: https://github.com/andela/mag-ah-frontend.git
 $ Navigate to `mag-ah-frontend`
 $ Chekout to branch `ft-share-articles-#159965275`
 $ Run npm install
 $ Run npm start
 
#### Any background context you want to provide?

screen shot 2018-10-29 at 17 00 15

Twitter
screen shot 2018-10-29 at 17 01 58

Facebook
screen shot 2018-10-29 at 17 09 45

Email
screen shot 2018-10-29 at 17 05 39
#### What are the relevant pivotal tracker stories?
[#159965275](https://www.pivotaltracker.com/story/show/159965275)

### CODE REVIEW

**Comment by pabishai** [comment]
Change the title of the PR to follow Andela best practices

**Comment by pabishai** [comment]
@mikenthiwa The buttons are too big in my opinion. The PO might have a different opinion so you can ask him as well.

**Comment by micahoriaso** [comment]
@mikenthiwa you PR is looking to merge into master. Change it to develop.

**Comment by mikenthiwa** [comment]
@PAbishai Done

**Comment by mikenthiwa** [comment]
@micahoriaso Done

File: .env-sample [hunk 1]
```diff
@@ -4,4 +4,4 @@ REACT_APP_FACEBOOK_KEY=""
 REACT_APP_FACEBOOK_SECRET=""
 REACT_APP_TWITTER_KEY=""
 REACT_APP_TWITTER_SECRET=""
-REACT_APP_BASE_URL=herokuBackendApiUrl.com/api
+REACT_APP_BASE_URL=
```
Comment (micahoriaso) [inline .env-sample]: Please return the backend url sample

File: public/index.html [hunk 2]
```diff
@@ -16,6 +16,14 @@
   <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css" integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU"
     crossorigin="anonymous">
   <!--
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json">
+    <link href="https://fonts.googleapis.com/css?family=Raleway" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Vidaloka" rel="stylesheet">
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css" integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU" crossorigin="anonymous">
+    <link href="https://fonts.googleapis.com/css?family=Bai+Jamjuree" rel="stylesheet">
```
Comment (micahoriaso) [inline public/index.html]: @mikenthiwa  Are you using these fonts?

File: src/components/articles/share.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
```
Comment (micahoriaso) [inline src/components/articles/share.jsx]: This is a stateless component please put it under /views folder

File: src/__tests__/jest/actions/article.test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import GetArticle from "../../../redux/actions/article";
+import { VIEW_ARTICLE } from "../../../redux/action_types";
+import article from "../../mock/article";
+
+const middleware = [thunk];
+const mockStore = configureMockStore(middleware);
+let store;
+
+describe("Article Action tests", () => {
+  let mock;
```
Comment (kimotho-njoki) [inline src/__tests__/jest/actions/article.test.js]: Why is mock declared here and not on line 14?

File: src/__tests__/jest/reducers/articleReducer.test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const initialState = {
+  viewArticle: {
+    article: {},
+    error: {}
+  }
+};
+
+const action = { payload: {} };
+
+describe("Article reducer test", () => {
```
Comment (kimotho-njoki) [inline src/__tests__/jest/reducers/articleReducer.test.js:14]: Why have two `describes` testing the exact same reducer?

File: src/__tests__/jest/reducers/articleReducer.test.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  it("should return initial test when no test", () => {
```
Comment (kimotho-njoki) [inline src/__tests__/jest/reducers/articleReducer.test.js]: `should return initial test` or `initial state`?

File: src/__tests__/jest/reducers/articleReducer.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    expect(articleReducer(initialState, action)).toEqual(initialState);
+  });
+});
+
+describe("Article reducer test", () => {
+  it("should handle article reducer", () => {
```
Comment (kimotho-njoki) [inline src/__tests__/jest/reducers/articleReducer.test.js]: Make this more descriptive. What exactly are we handling?

File: src/__tests__/jest/reducers/articleReducer.test.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    action.type = VIEW_ARTICLE;
+    expect(articleReducer(initialState.viewArticle, action).article).toEqual(
+      action.payload
+    );
+  });
```
Comment (kimotho-njoki) [inline src/__tests__/jest/reducers/articleReducer.test.js:24]: Extra line

**Comment by kimotho-njoki** [comment]
@mikenthiwa handle my comments as well as @micahoriaso's

File: src/__tests__/jest/reducers/articleReducer.test.js (hunk 5 above)
Comment (mikenthiwa) [reply src/__tests__/jest/reducers/articleReducer.test.js:14]: I have removed one of the describes

File: public/index.html (hunk 2 above)
Comment (mikenthiwa) [reply public/index.html]: No, I am not using them. I have removed them

File: src/__tests__/jest/actions/article.test.js (hunk 4 above)
Comment (mikenthiwa) [reply src/__tests__/jest/actions/article.test.js]: @kimotho-njoki I have moved the variable mock to line 14

File: .env-sample (hunk 1 above)
Comment (mikenthiwa) [reply .env-sample]: @micahoriaso Done

File: src/__tests__/jest/reducers/articleReducer.test.js (hunk 8 above)
Comment (mikenthiwa) [reply src/__tests__/jest/reducers/articleReducer.test.js:24]: @kimotho-njoki Added the extra line

File: src/__tests__/jest/reducers/articleReducer.test.js (hunk 7 above)
Comment (mikenthiwa) [reply src/__tests__/jest/reducers/articleReducer.test.js]: @kimotho-njoki I have made it more descriptive
We are handling when there is a single article

**Review by micahoriaso — APPROVED** [review]

**Review by micahoriaso — APPROVED** [review]

**Pull request merged**